### PR TITLE
Optimize move task

### PR DESCRIPTION
**Pull request by MortenTobiasNielsen**
It is a little faster but not a fast as I would have hoped. I think we need an in-memory representation of the buildings grid (like task_data_to_save) to be able to optimize it further.

### CODE REVIEW

File: cMain.cpp [hunk 1]
```diff
@@ -3832,8 +3840,9 @@ bool cMain::check_buildings_grid() {
 		}
 
 	} else if (building_task == "Take" || building_task == "Put") {
-		for (int i = 0; i < std::stoi(building_amount_of_buildings); i++) {
-			if (from_into != "Wreck" && !find_building()) {
+		building_amount_of_buildings_int = std::stoi(building_amount_of_buildings);
+		for (int i = 0; i < building_amount_of_buildings_int; i++) {
+			if (from_into != "Wreck" && !find_building(building_amount_of_buildings_int)) {
 				wxMessageBox("Building location doesn't exist.\n1. Please use exactly the same coordinates as you used to build \n2. Check that you have not removed the building(s)\n3. Check that you are not putting this task before the Build task", "Please use the same coordinates");
```
Comment (theis999) [inline cMain.cpp:3846]: Didn't you add an error msg handler?

File: cMain.cpp [hunk 2]
```diff
@@ -3842,8 +3851,7 @@ bool cMain::check_buildings_grid() {
 		}
 
 	} else if (building_task == "Launch") {
-		building_amount_of_buildings = "1";
-		if (!find_building()) {
+		if (!find_building(1)) {
 			wxMessageBox("Building location doesn't exist.\n1. Please use exactly the same coordinates as you used to build \n2. Check that you have not removed the building(s)\n3. Check that you are not putting this task before the Build task", "Please use the same coordinates");
```
Comment (theis999) [inline cMain.cpp:3855]: Didn't you add an error msg handler?

**Review by theis999 — APPROVED** [review]
👍

File: cMain.cpp (hunk 1 above)
Comment (MortenTobiasNielsen) [reply cMain.cpp:3846]: What do you mean?

File: cMain.cpp (hunk 2 above)
Comment (MortenTobiasNielsen) [reply cMain.cpp:3855]: What do you mean?

Comment (theis999) [reply cMain.cpp:3855]: For "wxMessageBox" i seem to recall you made a function to handle showing the error box. Maybe that was only related to the save file?

Comment (MortenTobiasNielsen) [reply cMain.cpp:3855]: Ohh yeah, you are right, but it was specific to save file.

**Pull request merged**